lib/pages/main/wallet_contents/add_wallet_connect/add_wallet_connect.dart

```diff
@@ -0,0 +1,333 @@
+              : [];
+          wcPairingNamespaces = args.params.generatedNamespaces;
+
+          List<String?> requiredNetworkIDs = [];
+          args.params.requiredNamespaces.forEach((key, value) {
+            requiredNetworkIDs.addAll(value.chains?.toList() ?? []);
+          });
+          for (var network in requiredNetworkIDs) {
+            if (network != null && network != Config.walletConnectChainId) {
+              notSupportedNetworks.add(network);
+            }
+          }
+        }
+        final invalidApp = args?.verifyContext?.validation.invalid;
+        final unknown = args?.verifyContext?.validation.unknown;
+        final scamApp = args?.verifyContext?.validation.scam;
+
+        if (pairingTimer != null) pairingTimer!.cancel();
+        bool? userhasConfirmed =
+            await Navigator.of(context).push(MaterialPageRoute<bool>(
+                builder: (BuildContext context) {
+                  return Pair(
+                    pairingId: wcPairingId!,
+                    pairingEvents: wcPairingEvents,
+                    pairingMethods: wcPairingMethods,
+                    pairingNamespaces: wcPairingNamespaces,
+                    pairingMetadata: wcPairingMetadata,
+                    unsupportedNetowrks: notSupportedNetworks,
+                    domainType: scamApp == true
+                        ? DomainType.scam
+                        : invalidApp == true
+                            ? DomainType.mismatch
+                            : unknown == true
+                                ? DomainType.unknown
+                                : DomainType.valid,
+                  );
+                },
+                fullscreenDialog: true));
+
+        if (userhasConfirmed != null && userhasConfirmed) {
+          //Accepted
+          try {
+            if (mounted) {
+              Navigator.of(context).pop();
+              final l10n = l10nOf(context);
+              _globalSnackBar.show(l10n.wcConnectionApproved);
+            }
+          } catch (e) {
+            _globalSnackBar.showError(e.toString());
+            setState(() {
+              isLoading = false;
+              if (pairingTimer != null) pairingTimer!.cancel();
+            });
+          }
+        } else if (userhasConfirmed == null) {
+          //Rejected
+          setState(() {
+            isLoading = false;
+            if (pairingTimer != null) pairingTimer!.cancel();
+          });
+          await walletConnectService.web3Wallet?.rejectSession(
+              id: wcPairingId!,
+              reason: Errors.getSdkError(Errors.USER_REJECTED).toSignError());
+
+          if (mounted) {
+            Navigator.of(context).pop();
+            final l10n = l10nOf(context);
+            _globalSnackBar.showError(l10n.wcConnectionRejected);
+          }
+        } else {
+          // false : expired
+          setState(() {
+            isLoading = false;
+            if (pairingTimer != null) pairingTimer!.cancel();
```

Comment:
it looks this is already cancelled in line 139

Reply:
This may be obsolete after upgrade to reown. Check comment in ahmeds answer. Will remove it for now